internal/cache/cache_manager.go

```diff
@@ -61,6 +62,14 @@ func (c *cacheManagerImpl) ConvertToFlagStruct(loadedFlags []byte, fileFormat st
 }
 
 func (c *cacheManagerImpl) UpdateCache(newFlags map[string]dto.DTO, log *fflog.FFLogger) error {
+	return c.updateCache(newFlags, log, false)
+}
+
+func (c *cacheManagerImpl) UpdateCacheAndNotify(newFlags map[string]dto.DTO, log *fflog.FFLogger) error {
+	return c.updateCache(newFlags, log, true)
+}
```

Comment:
Do we need those functions or should we just keep `UpdateCache` with the new parameter?

Reply:
### Conclusion

Sure, I think I can change the code to do that. Let me change my code.
Given that it's an internal interface that's not meant to be used by anything outside of this repo's code, I think we don't need to care so much for readability (or method clarity) as much as I initially thought.

### My initial thought

I just thought that, given that `CacheManager` is (technically) defined as an interface, it would be beneficial to have a separate method to clearly indicate what the method is doing on the inside, rather than having an argument that doesn't really convey anything to the reader of the code.

What I mean is, if I have a code that calls this method defined in the interface, it would be (slightly) easier for the readers to understand what's going on when written like
```go
err = cache.UpdateCacheAndNotify(newFlags, config.internalLogger)
// do other stuff...
```
rather than
```go
err = cache.UpdateCache(newFlags, config.internalLogger, true)
// do other stuff...
```
because the value of `true` doesn't really convey anything just by reading it, whereas the method name `UpdateCacheAndNotify` clearly tells that it's gonna "notify something".